### PR TITLE
Merge external splicing counts

### PR DESCRIPTION
**Pull request by c-mertes**
This is the fresh take on the merging of external splicing counts. #169

### CODE REVIEW

File: drop/modules/aberrant-splicing-pipeline/Counting/01_4_countRNA_nonSplitReads_merge.R [hunk 1]
```diff
@@ -50,3 +52,5 @@ nonSplitCounts <- getNonSplitReadCountsForAllSamples(fds=fds,
                                                      longRead=params$longRead)
 
 message(date(), ":", dataset, " nonSplit counts done")
+
+file.create(snakemake@output$done)
```
Comment (vyepez88) [inline drop/modules/aberrant-splicing-pipeline/Counting/01_4_countRNA_nonSplitReads_merge.R]:
```suggestion
file.create(snakemake@output$done)

```

File: drop/modules/aberrant-splicing-pipeline/Counting/03_filter_expression_FraseR.R [hunk 2]
```diff
@@ -27,23 +29,44 @@ opts_chunk$set(fig.width=12, fig.height=8)
...
+# 
 fds <- filterExpressionAndVariability(fds, 
-                        minExpressionInOneSample = minExpressionInOneSample,
-                        minDeltaPsi = minDeltaPsi,
-                        filter=FALSE)
-devNull <- saveFraserDataSet(fds)
+        minExpressionInOneSample = minExpressionInOneSample,
+        minDeltaPsi = minDeltaPsi,
+        filter=FALSE)
+fds <- saveFraserDataSet(fds)
```
Comment (nickhsmith) [inline drop/modules/aberrant-splicing-pipeline/Counting/03_filter_expression_FraseR.R]: if using external counts save a new copy of the fds object
else use a symlink of the fds object 

use the new copy/link in future processing

**Review by vyepez88 — DISMISSED** [review]

**Review by vyepez88 — DISMISSED** [review]

File: docs/source/output.rst [hunk 3]
```diff
@@ -0,0 +1,129 @@
+Results and Output of DROP
+===========================
+
+DROP is intended to help researchers use RNA-Seq data in order to detect genes with aberrant expression,
+aberrant splicing and mono-allelic expression. By simplifying the workflow process we hope to provide
+easy to read and interpret html files and output files. This section is dedicated to explaining the relevant
+results files. We will use the results of the ``demo`` to explain the files generated.::
```
Comment (c-mertes) [inline docs/source/output.rst]:
```suggestion
results files. We will use the results of the ``demo`` to explain the files generated by the following commands:
```

File: docs/source/output.rst [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    mkdir drop_demo
+    cd drop_demo
+    drop demo
+    
+    snakemake -c1
+
+Aberrant Expression
++++++++++++++++++++
+
+html file
```
Comment (c-mertes) [inline docs/source/output.rst]:
```suggestion
HTML file
```

File: docs/source/output.rst [hunk 5]
```diff
@@ -0,0 +1,129 @@
+Results and Output of DROP
+===========================
+
+DROP is intended to help researchers use RNA-Seq data in order to detect genes with aberrant expression,
+aberrant splicing and mono-allelic expression. By simplifying the workflow process we hope to provide
+easy to read and interpret html files and output files. This section is dedicated to explaining the relevant
```
Comment (c-mertes) [inline docs/source/output.rst]:
```suggestion
easy to read and interpret HTML files and output files. This section is dedicated to explaining the relevant
```

File: docs/source/output.rst [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
++++++++++++++++++++
+
+html file
+#########
+Looking at the resulting ``Output/html/drop_demo_index.html`` we can see the ``AberrantExpression`` 
+tab at the top of the screen. Following that the Overview tab contains links to the:  
+
+* Counting Summaries 
+    * For each aberrant expression group
+        * split of local vs external sample counts
```
Comment (c-mertes) [inline docs/source/output.rst]:
```suggestion
        * number of local vs external sample
```

File: docs/source/output.rst [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+        * QC relating to reads and size factors for each sample
+        * histograms relating to mean count distribution with different conditions
+        * information about the expressed genes within each sample and as a dataset
+* Outrider Summaries
+    * For each aberrant expression group
+        * the number of aberrantly expressed gene per sample
```
Comment (c-mertes) [inline docs/source/output.rst]:
```suggestion
        * the number of aberrantly expressed genes per sample
```

File: docs/source/output.rst [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+html file
+#########
+Looking at the resulting ``Output/html/drop_demo_index.html`` we can see the ``AberrantExpression`` 
+tab at the top of the screen. Following that the Overview tab contains links to the:  
+
+* Counting Summaries 
+    * For each aberrant expression group
+        * split of local vs external sample counts
+        * QC relating to reads and size factors for each sample
+        * histograms relating to mean count distribution with different conditions
```
Comment (c-mertes) [inline docs/source/output.rst]:
```suggestion
        * histograms showing the mean count distributions based on different conditions
```

File: docs/source/output.rst [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+        * information about the expressed genes within each sample and as a dataset
```
Comment (c-mertes) [inline docs/source/output.rst]:
```suggestion
        * expressed genes within each sample and as a whole in the dataset
```

File: docs/source/output.rst [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+* Outrider Summaries
+    * For each aberrant expression group
+        * the number of aberrantly expressed gene per sample
+        * how batch correction is done and the resulting lack of batch effects
```
Comment (c-mertes) [inline docs/source/output.rst]:
```suggestion
        * correlation between samples before and after the autoencoder correction
```

File: docs/source/output.rst [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+        * which samples contain outliers
+        * results table
+* Files
+    * OUTRIDER files for each aberrant expression group
+        * For each of these files you can follow the `OUTRIDER vignette for individual analysis <https://www.bioconductor.org/packages/devel/bioc/vignettes/OUTRIDER/inst/doc/OUTRIDER.pdf>`_. 
+    * tsv files
+        * For each aberrant expression group
+            * results.tsv
+                * this tsv file contains only the significant genes and samples that meet the cutoffs defined in the ``config.yaml`` for ``padjCutoff`` and ``zScoreCutoff``
```
Comment (c-mertes) [inline docs/source/output.rst]:
```suggestion
* Files (for each aberrant expression group)
    * OUTRIDER data files (RDS)
        * You can follow the `OUTRIDER vignette for further individual analysis <https://www.bioconductor.org/packages/devel/bioc/vignettes/OUTRIDER/inst/doc/OUTRIDER.pdf>`. 
    * results files (TSV)
                * the result file contains only the significant genes and samples that meet the cutoffs defined in the ``config.yaml`` for ``padjCutoff`` and ``zScoreCutoff``
```

File: docs/source/output.rst [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+html file
+##########
+Looking at the resulting ``Output/html/drop_demo_index.html`` we can see the ``AberrantSplicing`` 
+tab at the top of the screen. Following that the Overview tab contains links to the:  
+
+* Counting Summaries 
+    * For each aberrant splicing group
+        * split of local (from internal BAM files) vs external sample counts
+        * split of local vs merged with external sample splicing/intron counts
+        * comparison of local and external log mean counts
```
Comment (c-mertes) [inline docs/source/output.rst]:
```suggestion
        * comparison of local and external mean counts
```

File: docs/source/output.rst [hunk 13]
```diff
@@ -0,0 +1,129 @@
+Results and Output of DROP
+===========================
+
```
Comment (nickhsmith) [inline docs/source/output.rst:3]: follow slack comments

File: drop/demo/config_relative.yaml [hunk 14]
```diff
@@ -16,13 +16,14 @@ exportCounts:
     - v29
```
Comment (c-mertes) [inline drop/demo/config_relative.yaml:16]: Do we need to maintain it twice the file? In the code base and in the resource tar file?

File: drop/demo/sample_annotation_relative.tsv [hunk 15]
```diff
@@ -1,23 +1,25 @@
-RNA_ID	RNA_BAM_FILE	DNA_VCF_FILE	DNA_ID	DROP_GROUP	PAIRED_END	COUNT_MODE	COUNT_OVERLAPS	STRAND	HPO_TERMS	GENE_COUNTS_FILE	GENE_ANNOTATION	GENOME
+RNA_ID	RNA_BAM_FILE	DNA_VCF_FILE	DNA_ID	DROP_GROUP	PAIRED_END	COUNT_MODE	COUNT_OVERLAPS	STRAND	HPO_TERMS	GENE_COUNTS_FILE	GENE_ANNOTATION	GENOME	SPLICE_COUNTS_DIR
```
Comment (c-mertes) [inline drop/demo/sample_annotation_relative.tsv:1]: Same as above, do we need to maintain it twice?

File: drop/modules/aberrant-expression-pipeline/Counting/filterCounts.R [hunk 16]
```diff
@@ -36,5 +36,13 @@ ods <- filterExpression(ods, gtfFile=txdb, filter=FALSE,
 # add column for genes with at least 1 gene
 rowData(ods)$counted1sample = rowSums(assay(ods)) > 0
 
+has_external <- !(all(ods@colData$GENE_COUNTS_FILE == "") || is.null(ods@colData$GENE_COUNTS_FILE))
+if(has_external){
+    ods@colData$isExternal <- as.factor(ods@colData$GENE_COUNTS_FILE != "")
+}else{
```
Comment (nickhsmith) [inline drop/modules/aberrant-expression-pipeline/Counting/filterCounts.R:42]: comment to explain

File: drop/modules/aberrant-splicing-pipeline/Counting/01_1_countRNA_splitReads_samplewise.R [hunk 17]
```diff
@@ -6,13 +6,13 @@
 #'    - snakemake: '`sm str(tmp_dir / "AS" / "{dataset}" / "splitReads" / "{sample_id}.Rds")`'
 #'  params:
 #'   - setup: '`sm cfg.AS.getWorkdir() + "/config.R"`'
-#'   - workingDir: '`sm cfg.getProcessedDataDir() + "/aberrant_splicing/datasets"`'
+#'   - workingDir: '`sm cfg.getProcessedDataDir() + "/aberrant_splicing/datasets/fromBam"`'
```
Comment (c-mertes) [inline drop/modules/aberrant-splicing-pipeline/Counting/01_1_countRNA_splitReads_samplewise.R]: use ..../datasets/raw-local-{dataset} raw-{dataset} {dataset}

File: drop/modules/mae-pipeline/MAE/Results.R [hunk 18]
```diff
@@ -119,13 +120,19 @@ fwrite(res[MAE_ALT == TRUE & rare == TRUE], snakemake@output$res_signif_rare,
...
 res[MAE == TRUE, N_MAE := .N, by = ID]
 res[MAE == TRUE & MAE_ALT == FALSE, N_MAE_REF := .N, by = ID]
 res[MAE_ALT == TRUE, N_MAE_ALT := .N, by = ID]
 res[MAE == TRUE & MAE_ALT == FALSE & rare == TRUE, N_MAE_REF_RARE := .N, by = ID]
 res[MAE_ALT == TRUE & rare == TRUE, N_MAE_ALT_RARE := .N, by = ID]
 
 rd <- unique(res[,.(ID, N, N_MAE, N_MAE_REF, N_MAE_ALT, N_MAE_REF_RARE, N_MAE_ALT_RARE)])
+
+# rd contains duplicate entries for each ID. IE when MAE==F N_MAE for ID1 is both .N and 0
+# summarize these duplicates by taking the maximum of each column for each ID
```
Comment (nickhsmith) [inline drop/modules/mae-pipeline/MAE/Results.R]: clean

**Review by nickhsmith — CHANGES_REQUESTED** [review]

**Review by c-mertes — COMMENTED** [review]
Please adapt the code as discussed.

**Review by nickhsmith — DISMISSED** [review]

File: drop/demo/config_relative.yaml (hunk 14 above)
Comment (nickhsmith) [reply drop/demo/config_relative.yaml:16]: it's not in the resource tar

**Review by nickhsmith — APPROVED** [review]

**Pull request merged**